### ISSUE TITLE
 YOLOv9c-Seg Model: Only Boxes Detected During Inference

### ISSUE THREAD

**Issue by XuanTung4501** [ultralytics/ultralytics#18523]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

Hello guys,

I’m experiencing an issue with instance segmentation using Ultralytics version 8.3.54. Although my model has a segmentation head and my dataset includes segmentation annotations, inference results only output bounding boxes without the segmentation masks.

Details:
Training Setup:

I trained a YOLOv9c-Seg model from scratch for a car damage instance segmentation task.
During training, I ensured that my dataset was annotated with segmentation masks, and the visualize batch from Ultralytics confirmed the presence of segments in the dataset.
Inference Issue:

Despite setting task="segment" during inference, the results only include bounding boxes with no segmentation masks.
The model's architecture does include a segmentation head.

#### visualize the infernece result:
![mask](https://github.com/user-attachments/assets/437e8881-b701-447e-92ff-4f92f4a9264c)
#### mask result (xy is empty):
![maslk1](https://github.com/user-attachments/assets/bb8ecdc3-d561-4a5e-ad65-e95b1cb62ab0)
#### training result:
![r](https://github.com/user-attachments/assets/553625da-2067-4de0-be3b-a627082bd0b8)
#### trainning params:
![train](https://github.com/user-attachments/assets/26ef5277-d8e3-4fe3-97e0-bcdd869e8881)
#### training batch 0 visualized:
![batch](https://github.com/user-attachments/assets/3afac4e7-98e8-44d2-9200-1fdcc8d045ee)

### Environment

Ultralytics 8.3.54 🚀 Python-3.12.8 torch-2.5.1+cu124 CUDA:0 (NVIDIA RTX A4000, 15890MiB)
Setup complete ✅ (16 CPUs, 93.6 GB RAM, 441.4/467.9 GB disk)

OS                  Linux-5.15.0-127-generic-x86_64-with-glibc2.31
Environment         Linux
Python              3.12.8
Install             pip
RAM                 93.59 GB
Disk                441.4/467.9 GB
CPU                 AMD Ryzen 7 5700G with Radeon Graphics
CPU count           16
GPU                 NVIDIA RTX A4000, 15890MiB
GPU count           2
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0

## Inference code
from ultralytics import YOLO
import matplotlib.pyplot as plt

model = YOLO("output/weights/last.pt")
result = model("minor-car-accidents.webp", conf=0.01, verbose=False)[0]
plt.figure(figsize=(15, 15))
plt.imshow(result.plot(labels=False, boxes=True, masks=True))
plt.axis("off")
plt.show()

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @XuanTung4501, thank you for your detailed report and for using Ultralytics 🚀! We appreciate the time you’ve taken to outline your setup and include so much context—it’s incredibly helpful.

We recommend checking out the [Docs](https://docs.ultralytics.com) for usage examples and troubleshooting, especially the [Predict](https://docs.ultralytics.com/modes/predict/) mode and [Segmentation Tasks](https://docs.ultralytics.com/tasks/segment/). These resources often answer the most common issues.

### 🐛 Bug Report
If this is indeed a bug, could you please confirm the following:
1. Provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) that demonstrates the issue. This helps us isolate the root cause as quickly as possible.
2. Confirm that your environment is using the latest `ultralytics` release by upgrading it along with all dependencies:
   ```bash
   pip install -U ultralytics
   ```
3. If the issue persists after upgrading, try running your inference in a verified environment like Google Colab or Docker to rule out any environment-specific problems.  

### Verified Environments
YOLO models can be run seamlessly in any of the following official setups:
- 🟢 Free GPU Notebooks: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>
- **Docker Image**: For quick containerized setups, refer to our [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker"></a>.

Additionally, check our CI status for any potential ongoing backend issues:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg"></a>

An Ultralytics engineer will assist you soon to resolve this situation. Let us know if you have further questions or updates in the meantime. We’re here to help! 😊

**Comment by Y-T-G** [comment]
> Despite setting task="segment" during inference, the results only include bounding boxes with no segmentation masks.

It's supposed to be passed while loading the model, not during inference.

**Comment by Y-T-G** [comment]
> model = YOLO("output/weights/last.pt")

Pass it here

**Comment by XuanTung4501** [comment]
# Mask with strange shape
I have inferred from the trained images that some objects have masks and some don't. However, the masks look quite strange, almost like boxes. I think the segment head weights are not good on my dataset. If that is the case, how should I finetune? Adjust the mask_ratio parameter for example. If not, what is the reason? Please give me your opinion.
### Visualized masks
![maskkkkk](https://github.com/user-attachments/assets/cbc7d13b-c390-4ab1-b2f6-f588f677687c)
@Y-T-G

**Comment by Y-T-G** [comment]
Can you show the output from Ultralytics? This seems like your own custom visualization.

**Comment by XuanTung4501** [comment]
I have resolved the problem; I appreciate your assistance! @Y-T-G
